Azure-Samples/communication-services-ios-calling-hero
# [Refactor] Use AppSettings.xcconfig to configure the app

## Purpose
The hero app required making changes to the AppSettings.plist file in order to set up AAD and the token URL endpoint. This could cause potential issues as the file must be changed. 
The change uses an optional AppSettings.xcconfig file, which is used to provide the configuration directly. This is also in line with the UI library.

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code
Pull down the code and create an AppSettings.xcconfig file at the top level. This should contain the following values, with substitutions made appropriately.

```
displayName = <if-wanted>
teamsUrl = <team-url-if-wanted>
groupCallUuid = <group-call-id-if-wanted>

communicationTokenFetchUrl = <your-endpoint>
aadClientId = <your-client-id>
aadTenantId = <your-tenant-id>
```
* Test the code

## What to Check
* Ensure that the presence of the aadClientId triggers the authentication
* Ensure that the display name populates properly
* Ensure that the UUID value populates properly for group calls
* Ensure that the teams URL populates properly for teams calls
